packages/fxa-content-server/server/lib/amplitude.js

```diff
@@ -396,26 +397,70 @@ const transform = initialize(SERVICES, EVENTS, FUZZY_EVENTS);
 module.exports = receiveEvent;
 
 function receiveEvent(event, request, data) {
-  if (amplitude.disabled || !event || !request || !data) {
+  if (!event || !request || !data) {
     return;
   }
 
-  const userAgent = ua.parse(request.headers['user-agent']);
+  if (amplitude.rawEvents) {
+    const rawEvent = {
+      event,
+      context: {
+        eventSource: 'content',
+        version: VERSION,
+        emailTypes: EMAIL_TYPES,
+        userAgent: request.headers['user-agent'],
+        ..._.pick(data, [
+          'deviceId',
+          'devices',
+          'emailDomain',
+          'emailSender',
+          'emailService',
+          'entrypoint_experiment',
+          'entrypoint_variation',
+          'entrypoint',
+          'experiments',
+          'flowBeginTime',
+          'flowId',
+          'lang',
+          'location',
+          'marketingOptIn',
+          'newsletters',
+          'planId',
+          'productId',
+          'service',
+          'syncEngines',
+          'templateVersion',
+          'uid',
+          'userPreferences',
+          'utm_campaign',
+          'utm_content',
+          'utm_medium',
+          'utm_source',
+          'utm_term',
+        ]),
+      },
+    };
+    logger.info('rawAmplitudeData', rawEvent);
```

Comment:
In local testing with a login flow I get

```
rawAmplitudeData {"event":{"type":"signin.submit","offset":7874,"time":1585762092238,"flowTime":7848},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"none","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"flow.signin.attempt","offset":7875,"time":1585762092239,"flowTime":7849},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"none","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"error.signin.auth.103","offset":8142,"time":1585762092506,"flowTime":8116},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"none","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"flow.signin.submit","offset":80673,"time":1585762164995,"flowTime":80605},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"signin.submit","offset":80674,"time":1585762164996,"flowTime":80606},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"flow.signin.attempt","offset":80674,"time":1585762164996,"flowTime":80606},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"screen.signin-token-code","offset":81231,"time":1585762165553,"flowTime":81163},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"signin-token-code.submit","offset":90874,"time":1585762175197,"flowTime":90807},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"signin-token-code.success","offset":90922,"time":1585762175245,"flowTime":90855},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"flow.experiment.sendSms.US","offset":90999,"time":1585762175420,"flowTime":91030},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[{"choice":"sendSms","group":"US"},{"choice":"sendSmsHeader","group":"control"}],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"experiment.US.sendSms.enrolled","offset":90999,"time":1585762175420,"flowTime":91030},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[{"choice":"sendSms","group":"US"},{"choice":"sendSmsHeader","group":"control"}],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"flow.timing.clients.fetch","offset":91092,"time":1585762175513,"flowTime":"60"},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[{"choice":"sendSms","group":"US"},{"choice":"sendSmsHeader","group":"control"}],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"flow.experiment.sendSmsHeader.control","offset":91093,"time":1585762175514,"flowTime":91124},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[{"choice":"sendSms","group":"US"},{"choice":"sendSmsHeader","group":"control"}],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"experiment.control.sendSmsHeader.enrolled","offset":91093,"time":1585762175514,"flowTime":91124},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[{"choice":"sendSms","group":"US"},{"choice":"sendSmsHeader","group":"control"}],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
rawAmplitudeData {"event":{"type":"screen.signin.sms","offset":91115,"time":1585762175536,"flowTime":91146},"context":{"eventSource":"content","version":"1.164.0","emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0","deviceId":"000c98612bba456198c1fcf3b3c9ab34","emailDomain":"other","entrypoint_experiment":"none","entrypoint_variation":"none","entrypoint":"fxa_discoverability_native","experiments":[{"choice":"sendSms","group":"US"},{"choice":"sendSmsHeader","group":"control"}],"flowBeginTime":1585762084390,"flowId":"63ca2e8eb8b13439f1991044341a8cce8155f81b4cf6a73491fcc001f46dbf5e","lang":"en","location":{},"service":"sync","syncEngines":[],"uid":"dd9870a3207c48b7bef84febf901bdbf","userPreferences":{},"utm_campaign":"none","utm_content":"none","utm_medium":"none","utm_source":"none","utm_term":"none"}}
```

Do you know why `"emailTypes":{"complete-reset-password":"reset_password","complete-signin":"login","verify-email":"registration"}`. Those values seem unrelated to the flow.

Reply:
Offhand, I would assume one of the transforms is taking this and either filtering it, or transforming it.

Reply:
When the event is in the `email` group, the email types map is used to find the `email_type` event property.

Reply:
Gotcha, thanks for clearing up!